src/activity-log/activity-log-base-use-case.dto.ts

```diff
@@ -1,6 +1,31 @@
-import { ActivityLogDocumentDTO, ActivityLogObjectDataDTO, ActivityLogObjectIdDTO } from './dtos';
+import {
+  ActivityLogCommunityDTO,
+  ActivityLogDocumentDTO,
+  ActivityLogGroupDTO,
+  ActivityLogObjectDataDTO,
+  ActivityLogObjectIdDTO,
+  ActivityLogUserDTO,
+  ActivityPropChangedDTO,
+} from './dtos';
 import { ACTIVITY_EVENT_TYPES, ACTIVITY_LOG_USE_CASES, ACTIVITY_OBJECT_TYPES } from './enums';
 
+export class BasePayloadDTO<T = object> {
+  requestId: string;
+  rootUseCase: ACTIVITY_LOG_USE_CASES;
+  actor: ActivityLogUserDTO;
+  community: ActivityLogCommunityDTO;
```

Comment:
Còn user nữa á e

Reply:
user để làm gì vậy anh

Reply:
Hôm qua anh nghe em nói là BasePayloadDTO sẽ chứa tất cả các prop mà các use-case sẽ cần dùng đến á. 1 số use-case như add member to group, thì có actor, user, group á e

Reply:
với case add member to group thì member là đối tượng thay đổi và sẽ nằm trong currentState và originalState chứ không tách thành field độc lập ở bên ngoài ạ

Reply:
Huyền dự tính là dùng struct này cho các use-case mới á. Nhưng trước giờ thì data của actor, user (member), thì nằm cùng cấp ở data.

Reply:
vậy merge ko breaking change đúng ko a? để em merge nè. Chứ em ko có follow chi tiết activity log nên ko control đc

Reply:
À khoan Đạt ơi. Còn vụ requestId nữa á.

Reply:
requestId em update document dto rồi, và vẫn xóa requestId trong payload đi